keps/sig-storage/20190530-pv-health-monitor.md

```diff
@@ -85,7 +88,9 @@ There could be conditions that cannot be reported by a CSI driver. One or more n
 The Kubernetes components that monitor the volumes and report events with volume health information include the following:
 
 * An external monitoring node agent on each kubernetes worker node.
+  * Node agent reports events on the pod object. If multiple pods are using the same volume, report events on all pods.
```

Comment:
Does this mean every node agent has to watch every pod? I think this could be quite expensive. I wonder if we need to consider direct kubelet support instead.

Reply:
Node agent watches PVC health.  If a PVC has an abnormal health condition, an event will to reported to the Pod that is using the PVC.

We can consider direct Kubelet support in Beta?

Reply:
That's fine but we should start the conversation now. I think it may be a complex topic as I think sig-node wanted to move monitoring out of kubelet.